src/sentry/testutils/cases.py

```diff
@@ -1232,6 +1238,169 @@ def _send_buckets(cls, buckets, entity):
         )
 
 
+class BaseMetricsLayerTestCase(ABC, BaseMetricsTestCase):
+    ENTITY_SHORTHANDS = {
+        "c": "counter",
+        "s": "set",
+        "d": "distribution",
+        "g": "gauge",
+    }
+
+    @abstractmethod
```

Comment:
I would prefer to use abstractproperty:
```
    @property
    @abstractmethod
```
it's depricated so https://docs.python.org/3/library/abc.html#abc.abstractproperty you can add just property. It provide you syntax sugar like:
```
self.now
```
instead of 
```
self.now()
```